core/src/main/java/com/linecorp/armeria/common/HttpResponse.java

```diff
@@ -768,7 +769,18 @@ default HttpResponse mapHeaders(Function<? super ResponseHeaders, ? extends Resp
                 }
             }
             return obj;
-        });
+        })
+                .mapError(obj -> {
+                    if (obj instanceof HttpResponseException) {
+                        final HttpResponseException httpResponseException = (HttpResponseException) obj;
+                        return HttpResponseException.of(
+                                httpResponseException.httpResponse().mapHeaders(function),
+                                httpResponseException.getCause()
+                        );
+                    } else {
+                        return obj;
+                    }
+                });
```

Comment:
> but I believe the case is already handled by HttpResponse.mapHeaders()

I thought we could remove `.mapError(obj -> ...)` and prefer to use `AbortedHttpResponse` to mutate the headers of `HttpResponseException`.

Reply:
Aha, so that's what you meant! That makes sense. I'll fix it